### PR TITLE
[Microservice] Port Mgr Support L3 Routing with Routing Rules

### PR DESCRIPTION
**Pull request by xieus**
This PR proposes the following changes to Port Manager and Route Manager to enable L3 routing:
- PM starts using response from GetConnectedSubnet API from RM, and passing down associated router info (including routing table and routing rule)
- RM GetConnectedSubnet API refactor and issue fixes:
	- Change ConnectedSubnetsWebResponse to include a list of SubnetEntities, instead of SubnetIds so that PM could pass down all connected subnets to DPM
    - Router type fix (should be neutron_router, instead of neutron)
- PM issue fix:
    - Fix remote ip in neighbor table to be the ip address of neighbor port, instead of the neighbor port id.

### CODE REVIEW

**Comment by xieus** [comment]
@chenpiaoping Please take a look at this PR.

File: services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java [hunk 1]
```diff
@@ -425,83 +425,131 @@ public RoutesToNeutronWebResponse removeRoutesToNeutronRouter(String routerid, R
...
+
+        Router router = this.routerDatabaseService.getByRouterId(subnetWebJson.getSubnet().getAttachedRouterId());
+        if (!this.ValidateRouter(router)) {
+            logger.log(Level.WARNING, "Validation failed for router | router :" + router);
+            return connectedSubnetsWebResponse;
+        }
+
+        String routeTableType = router.getNeutronRouteTable().getRouteTableType();
+        Map<String, String> gwPortToSubnetIdMap = new HashMap<>();
+        if (routeTableType.equals("neutron_router")) {
```
Comment (kevin-zhonghao) [inline services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java:447]: Should we take "neutron_router" as the constants format value?

**Review by kevin-zhonghao — APPROVED** [review]
LGTM

File: services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java (hunk 1 above)
Comment (xieus) [reply services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java:447]: Yes. This was a bug found through debugging.

**Review by cj-chung — APPROVED** [review]
LGTM

**Comment by xieus** [comment]
The ci-bot issue is transient. All test cases pass locally.

**Pull request merged**